pyuvdata/uvh5.py

```diff
@@ -0,0 +1,279 @@
+        header['Nbls'] = self.Nbls
+        header['Nblts'] = self.Nblts
+        header['Nfreqs'] = self.Nfreqs
+        header['Npols'] = self.Npols
+        header['Nspws'] = self.Nspws
+        header['Ntimes'] = self.Ntimes
+        header['antenna_names'] = self.antenna_names
+        header['antenna_numbers'] = self.antenna_numbers
+        header['uvw_array'] = self.uvw_array
+        header['vis_units'] = self.vis_units
+        header['channel_width'] = self.channel_width
+        header['time_array'] = self.time_array
+        header['freq_array'] = self.freq_array
+        header['integration_time'] = self.integration_time
+        header['lst_array'] = self.lst_array
+        header['polarization_array'] = self.polarization_array
+        header['spw_array'] = self.spw_array
+        header['ant_1_array'] = self.ant_1_array
+        header['ant_2_array'] = self.ant_2_array
+
+        # write out phasing information
+        header['phase_type'] = self.phase_type
+        if self.zenith_dec is not None:
+            header['zenith_dec'] = self.zenith_dec
+        if self.zenith_ra is not None:
+            header['zenith_ra'] = self.zenith_ra
+        if self.phase_center_ra is not None:
+            header['phase_center_ra'] = self.phase_center_ra
+        if self.phase_center_dec is not None:
+            header['phase_center_dec'] = self.phase_center_dec
+        if self.phase_center_epoch is not None:
+            header['phase_center_epoch'] = self.phase_center_epoch
+
+        # write out optional parameters
+        if self.antenna_positions is not None:
+            header['antenna_positions'] = self.antenna_positions
+        if self.dut1 is not None:
+            header['dut1'] = self.dut1
+        if self.earth_omega is not None:
+            header['earth_omega'] = self.earth_omega
+        if self.gst0 is not None:
+            header['gst0'] = self.gst0
+        if self.rdate is not None:
+            header['rdate'] = self.rdate
+        if self.timesys is not None:
+            header['timesys'] = self.timesys
+        if self.x_orientation is not None:
+            header['x_orientation'] = self.x_orientation
+        if self.antenna_diameters is not None:
+            header['antenna_diameters'] = self.antenna_diameters
+        if self.uvplane_reference_time is not None:
+            header['uvplane_reference_time'] = self.uvplane_reference_time
+
+        # write out extra keywords if it exists and has elements
+        if self.extra_keywords:
+            extra_keywords = header.create_group("extra_keywords")
+            for k in self.extra_keywords.keys():
+                extra_keywords[k] = self.extra_keywords[k]
+
+        # write out history
+        header['history'] = self.history
+
+        # write out data and flags
+        # TODO: add filter options for visdata (bitshuffle), flags, and
+        #     nsample_array (lzf compression)
+        dgrp = f.create_group("Data")
```

Comment:
Jack might get into trouble here. Having to buffer an entire file in memory before writing it might be problematic.  Do you have an idea of how to go about doing streaming writes?

Reply:
Yes, I believe I know how to accomplish this, and was going to implement it with the "partial read" functionality. If we need the "partial write" sooner, I can prioritize that.